### PR TITLE
Setup Github action and pr flow

### PR DESCRIPTION
**Pull request by simplesagar**
This PR setups an automated PR workflow (Github action) that will regenerate this SDK and publish to package managers every night at midnight.I got the latest openapi spec for circle-ci v2 from this path: blob:https://circleci.com/99952f18-d475-4831-96e4-146d36c4ff79. It doesn't look like a stable URL so if one does exist we can provide update `openapi_doc_location` in the github workflow config to point to that instead of the local file path. 
Speakeasy docs for Github action setup: https://docs.speakeasyapi.dev/docs/tutorials/gh-actions-sdk

Also regenerated SDK using CLI version `0.19.0`

Followups needed from @roopakv. The following needed to be added Github secrets on this repo
- Invited you to a speakeasy workspace [circleci-dev.](https://app.speakeasyapi.dev/workspaces/cld9g3bs200003b6bn42v3ir3/apikeys) Create an API key in the UI and add to GH secrets as `SPEAKEASY_API_KEY`
- if you want the action to publish to `npm` please add a npm secret as Github secret`NPM_TOKEN` . I've currently configured the packagename as `@swissknife/circle-ts` but you can change that in `gen.yaml`

### CODE REVIEW

File: gen.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 typescript:
-  packagename: circle-ts
+  packagename: '@swissknife/circle-ts'
```
Comment (roopakv) [inline gen.yaml:2]:
```suggestion
  packagename: 'circleci-v2-sdk'
```

**Review by roopakv — CHANGES_REQUESTED** [review]

**Comment by simplesagar** [comment]
updated the package name but it doesn't have a NPM organisation prefix

**Comment by simplesagar** [comment]
@roopakv ok if i merge?

**Comment by roopakv** [comment]
yup merge. and no need org prefix

**Comment by roopakv** [comment]
ill add the tokens in an hour

**Pull request merged**